### PR TITLE
DG: Update Scope, NFR, Glossary

### LINKED ISSUE #33: Add non-functional requirements to DG

### LINKED ISSUE #34: Add glossary to DG

### LINKED ISSUE #40: Add Product Scope to DG

### PR DESCRIPTION
**Pull request by Berted**
Resolves #33, resolves #34, resolves #40.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -257,14 +257,17 @@ _{Explain here how the data archiving feature will be implemented}_
 
 **Target user profile**:
 
-* has a need to manage a significant number of contacts
+* work as teaching assistants
```
Comment (njxue) [inline docs/DeveloperGuide.md]: I think "works as a teaching assistant" to stay consistent with singular nouns.
Maybe could also end them with periods to stay consistent with the "Value propositin" section.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -257,14 +257,17 @@ _{Explain here how the data archiving feature will be implemented}_
...
 * can type fast
 * prefers typing to mouse interactions
 * is reasonably comfortable using CLI apps
 
-**Value proposition**: manage contacts faster than a typical mouse/GUI driven app
-
+**Value proposition**: 
+- Fast management of students' grades over the typical GUI-driven app.
+- Easy navigation and batch processing with the help of filter and search functionality.
+- CSV file generation of student data.
```
Comment (njxue) [inline docs/DeveloperGuide.md]: Should bullet points be * instead?

**Review by njxue — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Berted) [reply docs/DeveloperGuide.md]: Resolved by https://github.com/AY2223S1-CS2103T-T12-1/tp/pull/43/commits/7bf4eabce0a90a98255d040d7787420f35391459.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Berted) [reply docs/DeveloperGuide.md]: Resolved by https://github.com/AY2223S1-CS2103T-T12-1/tp/pull/43/commits/7bf4eabce0a90a98255d040d7787420f35391459.

**Review by njxue — APPROVED** [review]

**Pull request merged**